test/uk/gov/hmrc/cataloguefrontend/PrototypePageSpec.scala

```diff
@@ -50,5 +51,91 @@ class PrototypePageSpec
+        .withAuthToken("Token token")
+        .get()
+        .futureValue
+
+      response.status shouldBe 200
+      response.body should not include("password-reset")
+    }
+
+    "display success message when password changed successfully" in {
+      setupChangePrototypePasswordAuthEndpoint(hasAuth = true)
+      serviceEndpoint(GET, "/api/v2/repositories", willRespondWith = (200, Some(JsonData.repositoryData("2fa-prototype"))))
+      serviceEndpoint(GET, "/api/v2/repositories/2fa-prototype", willRespondWith = (200, Some(prototypeDetailsData)))
+      serviceEndpoint(GET, "/api/jenkins-jobs/2fa-prototype", willRespondWith = (200, Some(jenkinsBuildData)))
+      serviceEndpoint(GET, "/pr-commenter/repositories/2fa-prototype/report", willRespondWith = (404, Some("")))
+
+      serviceEndpoint(
+        POST,
+        "/v1/SetHerokuPrototypePassword",
+        willRespondWith = (200, Some("""{ "success": true, "message": "ok" }""")),
+        givenJsonBody = Some("""{ "app_name": "2fa-prototype", "password": "password" }""")
+      )
+
+      val response = wsClient
+        .url(s"http://localhost:$port/prototype/2fa-prototype/change-password")
+        .withAuthToken("Token token")
+        .withHttpHeaders("Csrf-Token" -> "nocheck", "Content-Type" -> "application/x-www-form-urlencoded")
+        .post(Map("password" -> Seq("password")))
+        .futureValue
+
+      response.status shouldBe 200
+      response.body should include("password-change-success-msg")
+
+      wireMockServer.verify(1, postRequestedFor(urlPathEqualTo("/v1/SetHerokuPrototypePassword")))
+    }
+
+    "display error message when password change failed downstream" in {
+      setupChangePrototypePasswordAuthEndpoint(hasAuth = true)
+      serviceEndpoint(GET, "/api/v2/repositories", willRespondWith = (200, Some(JsonData.repositoryData("2fa-prototype"))))
+      serviceEndpoint(GET, "/api/v2/repositories/2fa-prototype", willRespondWith = (200, Some(prototypeDetailsData)))
+      serviceEndpoint(GET, "/api/jenkins-jobs/2fa-prototype", willRespondWith = (200, Some(jenkinsBuildData)))
+      serviceEndpoint(GET, "/pr-commenter/repositories/2fa-prototype/report", willRespondWith = (404, Some("")))
+
+      serviceEndpoint(
+        POST,
+        "/v1/SetHerokuPrototypePassword",
+        willRespondWith = (400, Some("""{ "code": "INVALID_PASSWORD", "message": "error" }""")),
+        givenJsonBody = Some("""{ "app_name": "2fa-prototype", "password": "password" }""")
+      )
+
+      val response = wsClient
+        .url(s"http://localhost:$port/prototype/2fa-prototype/change-password")
+        .withAuthToken("Token token")
+        .withHttpHeaders("Csrf-Token" -> "nocheck", "Content-Type" -> "application/x-www-form-urlencoded")
+        .post(Map("password" -> Seq("password")))
+        .futureValue
+
+      response.status shouldBe 400
+      response.body should include("password-change-error-msg")
```

Comment:
should we assert that the response includes the message as returned from the api?